src/component/calculator.js

```diff
@@ -1,64 +1,93 @@
+import React, { useState } from 'react';
 import '../styles/calc.css';
+import calculate from '../logic/calculate';
 
 function MyComponent() {
+  const [preScreen, setPreScreen] = useState('');
+  const [curScreen, setCurScreen] = useState('0');
+  const [operation, setOperation] = useState(null);
+  const [showResult, setShowResult] = useState(false);
+
+  const handleButtonClick = (buttonName) => {
+    let result = calculate({ total: preScreen, next: curScreen, operation }, buttonName);
+
+    if (buttonName === '=') {
+      result = calculate({ total: preScreen, next: curScreen, operation }, buttonName);
+      setPreScreen('');
+      setCurScreen(result.total || '0');
+      setOperation(null);
+      setShowResult(true);
+    } else {
+      setPreScreen(result.total || '');
+      setCurScreen(result.next || '0');
+      setOperation(result.operation || null);
+      setShowResult(false);
+    }
+  };
+
+  const displayScreen = showResult ? (curScreen || '0') : `${preScreen} ${operation || ''}`;
+
   return (
     <div className="calc-container">
-      <div className="calc-screen">0</div>
-      <button className="calc-btn  sign-btn" type="button">
+      <div className="calc-screen">
+        <div className="pre-screen">{displayScreen}</div>
+        {!showResult && <div className="cur-screen">{curScreen}</div>}
```

Comment:
- [ Required ] I think it would be better if you improve the user experience(UX) of your users when using the App. I noticed that when I enter a number e.g `45` then the operation button `X` or `+` there is a `zero` that is added alongside the arithmetic operator e.g. `45 X0`, I think this behavior is a bit confusing especially if the user wants to input another number after the arithmetic operator e.g `45 X 2`. 
 
Please see the screenshot below and rectify this, when I click on an operation only the first number and the arithmetic operator should be displayed on the screen.

## Your app 
https://github.com/mohashyne/math_magicians_app/assets/16392046/886cdbb0-5690-4d63-9f9f-7ccb125423f8